cardano-cli/src/Cardano/CLI/Environment.hs

```diff
@@ -18,18 +21,35 @@ import           Text.Read (readMaybe)
 data EnvCli = EnvCli
   { envCliNetworkId :: Maybe NetworkId
   , envCliSocketPath :: Maybe FilePath
+  , envCliAnyCardanoEra :: Maybe AnyCardanoEra
   }
 
 getEnvCli :: IO EnvCli
 getEnvCli = do
   mNetworkId <- getEnvNetworkId
   mSocketPath <- getEnvSocketPath
+  mCardanoEra <- getCardanoEra
 
   pure EnvCli
     { envCliNetworkId = mNetworkId
     , envCliSocketPath = mSocketPath
+    , envCliAnyCardanoEra = mCardanoEra
     }
 
+envCliAnyShelleyBasedEra :: EnvCli -> Maybe AnyShelleyBasedEra
+envCliAnyShelleyBasedEra envCli =
+  case envCliAnyCardanoEra envCli of
```

Comment:
You can use `Monad Maybe` here and get rid of one `case`:
```haskell
AnyCardanoEra era <- envCliAnyCardanoEra envCli
```

Reply:
Done!